pkg/controller/atlascustomrole/atlascustomrole_controller.go

```diff
@@ -0,0 +1,245 @@
+	}
+
+	if !valid {
+		return r.terminate(workflowCtx,
+			atlasCustomRole,
+			api.ResourceVersionStatus,
+			workflow.AtlasResourceVersionMismatch,
+			true,
+			fmt.Errorf("version of the resource '%s' is higher than the operator version '%s'", atlasCustomRole.GetName(), version.Version)), nil
+	}
+	workflowCtx.SetConditionTrue(api.ResourceVersionStatus).SetConditionTrue(api.ValidationSucceeded)
+
+	if !r.AtlasProvider.IsResourceSupported(atlasCustomRole) {
+		return r.terminate(workflowCtx, atlasCustomRole,
+			api.ProjectCustomRolesReadyType, workflow.AtlasGovUnsupported,
+			false,
+			fmt.Errorf("the %T is not supported by Atlas for government", atlasCustomRole)), nil
+	}
+
+	credentials, err := selectCredentials(workflowCtx.Context, r.Client, atlasCustomRole)
+	if err != nil {
+		return r.fail(req, err), nil
+	}
+
+	atlasSdkClient, _, err := r.AtlasProvider.SdkClient(workflowCtx.Context, credentials, workflowCtx.Log)
+	if err != nil {
+		return r.terminate(workflowCtx,
+			atlasCustomRole,
+			api.ProjectCustomRolesReadyType,
+			workflow.AtlasAPIAccessNotConfigured,
+			true,
+			fmt.Errorf("unable to create atlas client: %s", err.Error())), nil
+	}
+	workflowCtx.SdkClient = atlasSdkClient
+	if res := handleCustomRole(workflowCtx, r.Client, atlasCustomRole, r.ObjectDeletionProtection); !res.IsOk() {
+		return r.fail(req, fmt.Errorf("%s", res.GetMessage())), nil
+	}
+	return r.idle(workflowCtx), nil
+}
+
+func selectCredentials(ctx context.Context, k8sClient client.Client, akoRole *akov2.AtlasCustomRole) (*client.ObjectKey, error) {
+	switch {
+	// First, try the externalProjectID and it's credentials
+	case akoRole.Spec.ExternalProjectIDRef != nil:
+		if akoRole.Spec.ConnectionSecret == nil {
+			return nil, errors.New("the 'externalProjectIDRef' is set but the 'connectionSecret' is missing")
+		}
+		return &client.ObjectKey{Name: akoRole.Spec.ConnectionSecret.Name, Namespace: akoRole.GetNamespace()}, nil
+	// Try the external project ref
+	case akoRole.Spec.ProjectRef != nil:
+		// if the local credentials are set, use them
+		if akoRole.Spec.ConnectionSecret != nil {
+			return &client.ObjectKey{Name: akoRole.Spec.ConnectionSecret.Name, Namespace: akoRole.GetNamespace()}, nil
+		}
+		// otherwise, use those attached to the AtlasProject that is referenced by the externalProjectRef
+		project := &akov2.AtlasProject{}
+		err := k8sClient.Get(ctx,
+			client.ObjectKey{Name: akoRole.Spec.ProjectRef.Name, Namespace: akoRole.Spec.ProjectRef.Namespace}, project)
+		if err != nil {
+			return nil, errors.Wrap(err, "can not read credentials from AtlasProject")
+		}
+		if project.Spec.ConnectionSecret == nil || project.Spec.ConnectionSecret.Name == "" {
+			return nil, errors.Wrapf(err, "credentials for AtlasProject '%s' are not configured", project.GetName())
+		}
+		return &client.ObjectKey{Name: project.Spec.ConnectionSecret.Name, Namespace: project.Spec.ConnectionSecret.Namespace}, nil
```

Comment:
This particular bit here is wrong, because it will prevent users from falling back to global credentials, which is the default we need to mainaint. Instead I suggest:
```go
		return project.ConnectionSecretObjectKey(), nil
```

`project.ConnectionSecretObjectKey()` areadu has the logic needed here:
```go
func (p *AtlasProject) ConnectionSecretObjectKey() *client.ObjectKey {
	if p.Spec.ConnectionSecret != nil {
		var key client.ObjectKey
		if p.Spec.ConnectionSecret.Namespace != "" {
			key = kube.ObjectKey(p.Spec.ConnectionSecret.Namespace, p.Spec.ConnectionSecret.Name)
		} else {
			key = kube.ObjectKey(p.Namespace, p.Spec.ConnectionSecret.Name)
		}
		return &key
	}
	return nil
}
```